radixdlt/src/main/java/com/radixdlt/consensus/liveness/ProposalGenerator.java

```diff
@@ -0,0 +1,38 @@
+package com.radixdlt.consensus.liveness;
+
+import com.google.common.collect.Sets;
+import com.radixdlt.common.Atom;
+import com.radixdlt.consensus.QuorumCertificate;
+import com.radixdlt.consensus.Round;
+import com.radixdlt.consensus.Vertex;
+import com.radixdlt.consensus.VertexStore;
+import com.radixdlt.mempool.Mempool;
+import java.util.List;
+import java.util.Objects;
+import javax.inject.Inject;
+
+/**
+ * Logic for generating new proposals
+ */
+public final class ProposalGenerator {
+	private final Mempool mempool;
+	private final VertexStore vertexStore;
+
+	@Inject
+	public ProposalGenerator(
+		VertexStore vertexStore,
+		Mempool mempool
+	) {
+		this.vertexStore = Objects.requireNonNull(vertexStore);
+		this.mempool = Objects.requireNonNull(mempool);
+	}
+
+	// TODO: check that next proposal works with current vertexStore state
+	public Vertex generateProposal(Round round) {
+		QuorumCertificate highestQC = vertexStore.getHighestQC();
```

Comment:
As discussed, I have some doubts that this is the best way to get the QC to extend. Fine for now, we can revisit when I integrate my branch.